internal/comment/directive.go

```diff
@@ -0,0 +1,28 @@
+package comment
+
+import (
+	"go/ast"
+	"strings"
+)
+
+type Directive string
+
+const (
+	prefix                     = `//exhaustruct:`
+	DirectiveIgnore  Directive = prefix + `ignore`
+	DirectiveEnforce Directive = prefix + `enforce`
+)
+
+// HasDirective parses a directive from a given list of comments.
+// If no directive is found, the second return value is `false`.
+func HasDirective(comments []*ast.CommentGroup, expected Directive) bool {
+	for _, cg := range comments {
+		for _, commentLine := range cg.List {
+			if strings.HasPrefix(commentLine.Text, string(expected)) {
```

Comment:
Small optimization: well-written directives should come after other comments (VSCode seems to reorder directives this way), so traversal of each comment group in reverse order would catch directives a bit faster.